### PR TITLE
Issue solo 64 refresh page error

### PR DESCRIPTION
**Pull request by zfi**
Add helper function getTimestamp().
Refactor the OnUnload event handler to prevent the accidental overwrite of the projectData object.

### CODE REVIEW

**Comment by pjewald** [comment]
This is one more iteration to correct the random issues with page reload after project changes have occurred.

File: editor.js [hunk 1]
```diff
@@ -236,23 +247,28 @@ function initUploadModalLabels() {
...
 function checkLeave () {
     // Return if there is no project data
-    if (!projectData) {
+    if (! projectData) {
         return false;
     }
 
-    var currentXml = getXml();
-    var savedXml = projectData.code;
+    let currentXml = getXml();
```
Comment (pjewald) [inline editor.js]: Refactor to use ES6 conventions.

File: editor.js [hunk 2]
```diff
@@ -529,32 +545,36 @@ $(document).ready( () => {
...
+            //   the call to this.performance.now() to get the
+            //   timestamp was failing. Replacing that call with a
+            //   generic function getTimestamp() resolves the error.
+            // ------------------------------------------------------
+
+            // Store the current project into the localStore so that
+            // if the page is being refreshed, it will automatically
+            // be reloaded
+            // ------------------------------------------------------
+            let tempProject = {};
```
Comment (pjewald) [inline editor.js]: Create a new, temporary object and then copy the contents of the projectData object into the temp object. This prevents projectData from being overwritten.

File: editor.js [hunk 3]
```diff
@@ -529,32 +545,36 @@ $(document).ready( () => {
...
+            tempProject.code = getXml();
+            tempProject.timestamp = getTimestamp();
+            window.localStorage.setItem('localProject', JSON.stringify(tempProject));
         }
 
         if (checkLeave()) {
             e.preventDefault();     // Cancel the event
             e.returnValue = Blockly.Msg.DIALOG_CHANGED_SINCE;
             return Blockly.Msg.DIALOG_CHANGED_SINCE;
-        } else if (isOffline) {
```
Comment (pjewald) [inline editor.js]: This block is removed now that the calls to this.performance.now() have been removed.

**Comment by zfi** [comment]
Addresses issue parallaxinc/solo#64.

**Pull request merged**